block/block_cache.go

```diff
@@ -8,61 +8,58 @@ import (
 
 // BlockCache maintains blocks that are seen and hard confirmed
 type BlockCache struct {
-	blocks     map[uint64]*types.Block
-	hashes     map[string]bool
-	daIncluded map[string]bool
-	mtx        *sync.RWMutex
+	blocks     *sync.Map
+	hashes     *sync.Map
+	daIncluded *sync.Map
 }
 
 // NewBlockCache returns a new BlockCache struct
 func NewBlockCache() *BlockCache {
 	return &BlockCache{
-		blocks:     make(map[uint64]*types.Block),
-		hashes:     make(map[string]bool),
-		daIncluded: make(map[string]bool),
-		mtx:        new(sync.RWMutex),
+		blocks:     new(sync.Map),
+		hashes:     new(sync.Map),
+		daIncluded: new(sync.Map),
 	}
 }
 
-func (bc *BlockCache) getBlock(height uint64) (*types.Block, bool) {
-	bc.mtx.Lock()
-	defer bc.mtx.Unlock()
-	block, ok := bc.blocks[height]
-	return block, ok
+func (bc *BlockCache) getBlock(height uint64) (types.Block, bool) {
```

Comment:
is there a reason for changing the signature? Keeping the return type `*types.Block` will avoid a copy.

Reply:
The reason initially was just not knowing how to use `sync.Map` and how casting works, tried switching it back to a pointer, and that works as well. So fixed.